src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.salesforce.connector;
+
+import com.sforce.soap.partner.DescribeGlobalResult;
+import com.sforce.soap.partner.PartnerConnection;
+import com.sforce.ws.ConnectionException;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.batch.BatchSink;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.connector.BrowseDetail;
+import io.cdap.cdap.etl.api.connector.BrowseEntity;
+import io.cdap.cdap.etl.api.connector.BrowseEntityPropertyValue;
+import io.cdap.cdap.etl.api.connector.BrowseRequest;
+import io.cdap.cdap.etl.api.connector.Connector;
+import io.cdap.cdap.etl.api.connector.ConnectorContext;
+import io.cdap.cdap.etl.api.connector.ConnectorSpec;
+import io.cdap.cdap.etl.api.connector.ConnectorSpecRequest;
+import io.cdap.cdap.etl.api.connector.PluginSpec;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.plugin.common.ConfigUtil;
+import io.cdap.plugin.salesforce.SObjectDescriptor;
+import io.cdap.plugin.salesforce.SalesforceConnectionUtil;
+import io.cdap.plugin.salesforce.SalesforceConstants;
+import io.cdap.plugin.salesforce.SalesforceSchemaUtil;
+import io.cdap.plugin.salesforce.authenticator.AuthenticatorCredentials;
+import io.cdap.plugin.salesforce.plugin.SalesforceConnectorConfig;
+import io.cdap.plugin.salesforce.plugin.sink.batch.SalesforceBatchSink;
+import io.cdap.plugin.salesforce.plugin.sink.batch.SalesforceSinkConfig;
+import io.cdap.plugin.salesforce.plugin.source.batch.SalesforceBatchSource;
+import io.cdap.plugin.salesforce.plugin.source.batch.util.SalesforceSourceConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Salesforce Connector Plugin
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(SalesforceConstants.PLUGIN_NAME)
+@Description("Connection to access data in Salesforce SObject.")
+public class SalesforceConnector implements Connector {
```

Comment:
Is there a reason why we don't implement DirectConnector?

Reply:
Yes, we are not implementing sample method here that’s why Connector is being implemented here instead of DirectConnector.

Reply:
Why are we not implementing sampling? Is it not supported with Salesforce?

Reply:
In this, fetching of data is job based..that's why we are not implementing sampling here.

Reply:
If sampling is not supported, then the connector can't be used with wrangler so would prefer to support sampling if possible. Also, if sampling is not support you need to set `canSample(false)`.

Some questions:
1. What's the latency for fetching data by running a job? If it's not too much, we could still implement a DirectConnector.
2. Are there any other ways to fetch data that don't require submitting a job and polling for completion?

CC @yaojiefeng

Reply:
Making canSample(false) freezes the list of objects. User will not able to select the one from list and will not be redirected to the next.
Looking forward to find the best and optimised way of implementing sampling.